app/controllers/course/material/folders_controller.rb

```diff
@@ -44,8 +44,31 @@ def create_subfolder
     end
   end
 
+  def new_materials
+  end
+
+  def upload_materials
+    if @folder.update_attributes(files_params)
+      redirect_to course_material_folder_path(current_course, @folder),
+                  success: t('.success', name: @folder.name)
+    else
+      show_danger_flash(t('.failure', error: @folder.errors.full_messages.to_sentence))
```

Comment:
OK, but why taking out the  entire else branch is better ?  and I can't find a good name for it.